hoodie-common/src/main/java/com/uber/hoodie/common/table/log/LogBlockVersion.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2016 Uber Technologies, Inc. (hoodie-dev-group@uber.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.uber.hoodie.common.table.log;
+
+/**
+ * A set of feature flags associated with a log block format.
+ * Versions are changed when the log block format changes.
+ * TODO(na) - Implement policies around major/minor versions
+ */
+abstract class LogBlockVersion {
```

Comment:
in terms of abstraction, what I thought of was 

`HoodieLogBlockV1 extends LogBlockVersion` (this reads the blocks as it is today) 
`HoodieLogBlockV2 extends LogBlockVersion` (this is the new stuff) 

then we call `logBlockVersion.readBlock()` as I noted below.. Again, I can't think of anything else that will let us evolve the contents of a block itself (headers + content + footers) in a flexible way. For e.g if we decide to add indexes into data blocks we can just `HoodieLogBlockV3 extends HoodieLogBlockV2` and only override how data blocks are read..

we may also need a LogFormatVersion (something that is decided based on old or new magic) or that can be here as well?

Reply:
At the moment, this version is managing both.